File: solutions/typescript/01-tic-tac-toe/src/entities/Player.ts
export class Player {
    name: string;
    character: string;
    static Builder = class PlayerBuilder {
        private name!: string;
        private character!: string;


        setName(value: string): PlayerBuilder {
            this.name = value;
            return this;
        }

        setCharacter(value: string): PlayerBuilder {
            if (value == "_") {
                throw new Error("Underscore '_' is not a valid player character")
            }
            this.character = value;
            return this;
        }

        build(): Player {
            return new Player(this.name, this.character)
        }
    }


    private constructor(name: string, character: string) {
        this.name = name;
        this.character = character;
    }
}

File: solutions/java/03-snakeladder/src/main/java/com/scaler/entities/Player.java
package com.scaler.entities;

public class Player {
    private String name;
    private int position;

    public String getName() {
        return name;
    }

    public int getPosition() {
        return position;
    }

    private Player(String name, int position) {
        this.name = name;
        this.position = position;
    }

    public static class Builder {
        private String name;

        public Builder setName(String name) {
            this.name = name;
            return this;
        }

        Player build() {
            return new Player(name, 0);
        }
    }
    static Builder builder() {
        return new Builder();
    }
}
